pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -207,31 +230,38 @@ func (a *actuator) Delete(ctx context.Context, ms *seedmanagementv1alpha1.Manage
 
 	if seed != nil {
 		if seed.DeletionTimestamp == nil {
-			a.deletingInfoEventf(ms, "Deleting seed %s", ms.Name)
+			log.Info("Deleting seed", "seed", client.ObjectKey{Name: ms.Name})
+			a.recorder.Eventf(ms, corev1.EventTypeNormal, gardencorev1beta1.EventDeleting, "Deleting seed %s", ms.Name)
 			if err := a.deleteSeed(ctx, ms); err != nil {
 				return status, false, false, fmt.Errorf("could not delete seed %s: %w", ms.Name, err)
 			}
 		} else {
-			a.deletingInfoEventf(ms, "Waiting for seed %s to be deleted", ms.Name)
+			log.Info("Waiting for seed to be deleted", "seed", client.ObjectKey{Name: ms.Name}.String())
+			a.recorder.Eventf(ms, corev1.EventTypeNormal, gardencorev1beta1.EventDeleting, "Waiting for seed %q to be deleted", ms.Name)
 		}
+
 		return status, false, false, nil
 	}
 
 	if ms.Spec.Gardenlet != nil {
 		// Delete gardenlet from the shoot if it still exists and is not already deleting
 		gardenletDeployment, err := a.getGardenletDeployment(ctx, shootClient)
 		if err != nil {
-			return status, false, false, fmt.Errorf("could not get gardenlet deployment in shoot %s: %w", kutil.ObjectName(shoot), err)
+			return status, false, false, fmt.Errorf("could not get gardenlet deployment in shoot %s: %w", client.ObjectKeyFromObject(shoot).String(), err)
 		}
+
 		if gardenletDeployment != nil {
 			if gardenletDeployment.DeletionTimestamp == nil {
-				a.deletingInfoEventf(ms, "Deleting gardenlet from shoot %s", kutil.ObjectName(shoot))
-				if err := a.deleteGardenlet(ctx, shootClient, ms, seed, gardenletConfig, shoot); err != nil {
-					return status, false, false, fmt.Errorf("could delete gardenlet from shoot %s: %w", kutil.ObjectName(shoot), err)
+				log.Info("Deleting gardenlet from shoot", "shoot", client.ObjectKeyFromObject(shoot))
```

Comment:
You could add the `shoot` key-value pair to `log` after retrieving the shoot object instead of adding it to log statements individually.
BTW: the namespace of the shoot is already implicitly given, as `ManagedSeed` and `Shoot` must be in the same namepsace.